### PR TITLE
Add cnsrqlquery endpoint.

### PR DESCRIPTION
**Pull request by abhijitherekar**
## Description

This change adds the cnsrqlquery endpoint.

## Motivation and Context

Fixes: https://redlock.atlassian.net/browse/CNS-2554

### CODE REVIEW

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "2c44595e81f56afd4ca8b622a49323f63669e547",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "05e367cc99d3caf688aad4ec36f9b9d82284b493",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "7afed465d06baa81e9259b95d8e9aa71da77131c",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "05e367cc99d3caf688aad4ec36f9b9d82284b493",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "7afed465d06baa81e9259b95d8e9aa71da77131c",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "05e367cc99d3caf688aad4ec36f9b9d82284b493",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "1e7b719f425687ed06efc9fc7c2f21ee7a5ee3f0",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "05e367cc99d3caf688aad4ec36f9b9d82284b493",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "05e367cc99d3caf688aad4ec36f9b9d82284b493",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "c88a96d4907dea087db281c131df61bf00a1c24b",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "24918670a435f79fcfa14b994058501145c245ff",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "ff07bad3af4e1eb5cc1da674d47b54779250af6c",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "1f6758318bb3dc89565cb042f4185f1927f37671",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "ff07bad3af4e1eb5cc1da674d47b54779250af6c",
    "pipeline": "master"
  }
]
```

File: specs/cnsrqlquery.spec [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  description: A CNS endpoint which will generate the RQL query for the given alert-id.
+
+# Attributes
+attributes:
+  v1:
+  - name: alertID
+    exposed_name: alertID
+    description: ID of the query request.
+    type: string
+    exposed: true
```
Comment (t00f) [inline specs/cnsrqlquery.spec:17]: Based on the description, I believe the `alertID` is a mandatory property. If true, we should add `required: true` here

Comment (t00f) [inline specs/cnsrqlquery.spec:17]: Based on the description, I believe the `alertID` is a mandatory property. If true, we should add `required: true` here

File: specs/cnsrqlquery.spec [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    exposed_name: policyID
+    description: The policy for which the alert was generated.
+    type: string
+    exposed: true
+
+  - name: query
+    exposed_name: query
+    description: The rql query for the alert.
+    type: string
+    exposed: true
```
Comment (t00f) [inline specs/cnsrqlquery.spec:29]: Based on the description, I understand the query is the output computed. It should be `autogenerated: true` and `readonly: true` to make sure the user can't pass the query.

File: specs/cnsrqlquery.spec [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    exposed_name: alertID
+    description: ID of the query request.
+    type: string
+    exposed: true
+
+  - name: policyID
+    exposed_name: policyID
+    description: The policy for which the alert was generated.
+    type: string
+    exposed: true
```
Comment (t00f) [inline specs/cnsrqlquery.spec:23]: Is it required in the request body?

File: specs/cnsrqlquery.spec (hunk 1 above)
Comment (t00f) [reply specs/cnsrqlquery.spec:17]: I don't think we need `exposed_name` here. The name is the property being used.

**Review by t00f — COMMENTED** [review]
Please see my comments below

File: specs/cnsrqlquery.spec (hunk 1 above)
Comment (abhijitherekar) [reply specs/cnsrqlquery.spec:17]: yup, thats right. Will do that.

File: specs/cnsrqlquery.spec (hunk 2 above)
Comment (abhijitherekar) [reply specs/cnsrqlquery.spec:29]: thats right.

File: specs/cnsrqlquery.spec (hunk 3 above)
Comment (abhijitherekar) [reply specs/cnsrqlquery.spec:23]: I don't use it in the code but this filed is required as part of the request that was discussed with the PC team by @rramalinga .  in ticket: https://redlock.atlassian.net/browse/RLP-40744

Comment (abhijitherekar) [reply specs/cnsrqlquery.spec:23]: I don't use it in the code but this field is required as part of the request that was discussed with the PC team by @rramalinga .  in ticket: https://redlock.atlassian.net/browse/RLP-40744

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "cbb6f36046e38e46d152a24145a26dbf90dd0984",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "fd172365d60813bf9d60f77da7296391b824842e",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "a284d42d4713cc4f84a1322cebb3d4b8abd58f78",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "fd172365d60813bf9d60f77da7296391b824842e",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "0583404e0ec933dbab0f99678c0af4997bf94b63",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "fd172365d60813bf9d60f77da7296391b824842e",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "0583404e0ec933dbab0f99678c0af4997bf94b63",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "fd172365d60813bf9d60f77da7296391b824842e",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "2afd20944584aa5f9424c05cff9323a6bf384bc1",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "fd172365d60813bf9d60f77da7296391b824842e",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "fd172365d60813bf9d60f77da7296391b824842e",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "59b33aa524caf9834489d0c199c3d6accf1a344b",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "fd172365d60813bf9d60f77da7296391b824842e",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "55a0ee7222c6be2a81433aa8610b33c80c68599f",
    "pipeline": "master"
  }
]
```

**Review by rramalinga — DISMISSED** [review]

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "25d82bbd50c6164438e3c3a1929ed816d9abc6ce",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "95b0cd9f3b5a46a3f627435a6d5626772e1d97e1",
    "pipeline": "master"
  }
]
```

**Review by rramalinga — APPROVED** [review]

**Review by ericrpowers — COMMENTED** [review]
Please run `make format` to avoid the changes in root.spec that happened here. You may need to run `update-spectools` to get the latest.

@primalmotion We will need the vscode extension updated to have these changes there as well (as I noticed there is a difference between the two).

**Comment by abhijitherekar** [comment]
> Please run `make format` to avoid the changes in root.spec that happened here. You may need to run `update-spectools` to get the latest.
> 
> @primalmotion We will need the vscode extension updated to have these changes there as well (as I noticed there is a difference between the two).

I did the make format and it reverted those changes but, I do had the latest `update-spectools` so not sure what was missing.

**Review by ericrpowers — APPROVED** [review]

**Review by rramalinga — APPROVED** [review]

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cnsrqlquery",
    "component": "backend",
    "pr-id": "1292",
    "commit-sha": "a9e5d4e097cbf040d48967ca8354ff6180092dc0",
    "pipeline": "master"
  },
  {
    "project": "cnsrqlquery",
    "component": "gaia",
    "pr-id": "685",
    "commit-sha": "ad2d3a0f45d223c6de3455c28a4b41aecc435b05",
    "pipeline": "master"
  }
]
```

**Pull request merged**